cpp-lite/include/mxnet-lite/ndarray.h

```diff
@@ -0,0 +1,284 @@
+  */
+  static std::vector<NDArray> LoadList(const std::string &file_name) {
+    mx_uint out_size, out_name_size;
+    NDArrayHandle *out_arr;
+    const char **out_names;
+    CHECK_EQ(MXNDArrayLoad(file_name.c_str(), &out_size, &out_arr,
+                           &out_name_size, &out_names), 0);
+    std::vector<NDArray> array_list;
+    array_list.reserve(out_size);
+    for (mx_uint i = 0; i < out_size; ++i) {
+      array_list.push_back(NDArray(out_arr[i]));
+    }
+    return array_list;
+  }
+  /*!
+  * \brief Load map of NDArrays from binary file.
+  * \param file_name name of the binary file.
+  * \return a list of NDArrays.
+  */
+  static std::map<std::string, NDArray> LoadDict(const std::string &file_name) {
+    mx_uint out_size, out_name_size;
+    NDArrayHandle *out_arr;
+    const char **out_names;
+    CHECK_EQ(MXNDArrayLoad(file_name.c_str(), &out_size, &out_arr,
+                          &out_name_size, &out_names), 0);
+    CHECK_EQ(out_name_size, out_size)
+        << "Cannot load to dict because file was saved as list.";
+    std::map<std::string, NDArray> array_map;
+    for (mx_uint i = 0; i < out_size; ++i) {
+      array_map[out_names[i]] = NDArray(out_arr[i]);
+    }
+    return array_map;
+  }
+  /*!
+  * \brief save a map of string->NDArray to binary file.
+  * \param file_name name of the binary file.
+  * \param array_map a map from names to NDArrays.
+  */
+  static void Save(const std::string &file_name,
+                   const std::map<std::string, NDArray> &array_map) {
+    std::vector<NDArrayHandle> args;
+    std::vector<const char *> keys;
+    for (const auto &t : array_map) {
+      args.push_back(t.second.GetHandle());
+      keys.push_back(t.first.c_str());
+    }
+    CHECK_EQ(MXNDArraySave(file_name.c_str(), args.size(), args.data(),
+                           keys.data()), 0);
+  }
+  /*!
+  * \brief save a list of NDArrays to binary file.
+  * \param file_name name of the binary file.
+  * \param array_list a list of NDArrays.
+  */
+  static void Save(const std::string &file_name,
+                   const std::vector<NDArray> &array_list) {
+    std::vector<NDArrayHandle> args;
+    for (const auto &t : array_list) {
+      args.push_back(t.GetHandle());
+    }
+    CHECK_EQ(MXNDArraySave(file_name.c_str(), args.size(), args.data(),
+                           nullptr), 0);
+  }
+  /*!
+  * \brief construct with a none handle
+  */
+  NDArray() {
+    NDArrayHandle handle;
+    CHECK_EQ(MXNDArrayCreateNone(&handle), 0);
+    blob_ptr_ = std::make_shared<NDBlob>(handle);
+  }
+  /*!
+  * \brief construct with a NDArrayHandle
+  */
+  explicit NDArray(const NDArrayHandle &handle) {
+    blob_ptr_ = std::make_shared<NDBlob>(handle);
+  }
+  /*!
+  * \brief construct a new dynamic NDArray
+  * \param shape the shape of array
+  * \param constext context of NDArray
+  * \param delay_alloc whether delay the allocation
+  */
+  NDArray(const std::vector<mx_uint> &shape,
```

Comment:
better to have a class `Shape`? so that users can write something like `s.subshape(1, 3)` or pretty-print with `LOG << "shape = " << s` to debug